### PR TITLE
Draft 2019 Roadmap 

### PR DESCRIPTION
**Pull request by leostera**
We met today from ~7pm CET until almost ~11pm to:

1. Discuss the first RFC, which ended up in a few small amendments visible in this PR
2. Write a roadmap for the first initial phase of the project
3. Sketch a roadmap for the second

Thanks @ulrikstrid, @p1xelHer0, and @cem2ran

[Read the Rendered version here](https://github.com/reuniverse/reuniverse/blob/rfc/roadmap-2019/rfcs/0003-roadmap-2019.md)

### CODE REVIEW

File: rfcs/0003-roadmap-2019.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+- Name: roadmap-2019
+- Start Date: 2019-07-13
+- Issue/PR: (leave this empty)
+- Point of Contact: Leandro Ostera <leandro@src.technology>
+                    Pontus Nagy <pontusnagy@gmail.com>
+                    Ulrik Strid <ulrikstrid@outlook.com>
```
Comment (ulrikstrid) [inline rfcs/0003-roadmap-2019.md]: ulrik.strid@outlook.com

**Review by ulrikstrid — CHANGES_REQUESTED** [review]

**Review by ulrikstrid — DISMISSED** [review]

File: rfcs/0001-reuniverse.md [hunk 2]
```diff
@@ -166,6 +166,9 @@ concerns in a smooth fashion.
    sources of information (such as manually submitted packages), we will build
    an index of reason packages.
 
+   In the future, additional package sources could be included, such as Opam,
+   NixPkgs, or Entropy.
```
Comment (cem2ran) [inline rfcs/0001-reuniverse.md]:
```suggestion
   [Nixpkgs](https://github.com/NixOS/nixpkgs), or [Entropic](https://github.com/entropic-dev/entropic).
```

File: rfcs/0001-reuniverse.md [hunk 3]
```diff
@@ -166,6 +166,9 @@ concerns in a smooth fashion.
    sources of information (such as manually submitted packages), we will build
    an index of reason packages.
 
+   In the future, additional package sources could be included, such as Opam,
```
Comment (cem2ran) [inline rfcs/0001-reuniverse.md]:
```suggestion
   In the future, additional package sources could be included, such as [Opam](https://github.com/ocaml/opam-repository),
```

File: rfcs/0003-roadmap-2019.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+naively resembling a design system.
+
+**What has to be done**
+1. Rework the Search to use a less naive approach and include more metadata fields
+1. Define basics of the Design System
+1. Extend the index to include package origin information
+
+
+## Features #2: Package Documentation
+[feat-2-package-docs]: #feat-1-package-docs
```
Comment (ulrikstrid) [inline rfcs/0003-roadmap-2019.md]: typo: `#feat-1-package-docs` -> `#feat-2-package-docs`

File: rfcs/0003-roadmap-2019.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+packages in the index whose target platform is not `Unknown`.
+
+**What has to be done**
+1. Extract the sources for every package and verify their integrity
+1. Compile every downladed package version using the appropriate compiler (BuckleScript or OCaml)
+1. Generate documentation using the appropriate ODoc driver (BsDoc for `Web`, Dune for `Native` and `Universal` packages)
+
+
+## Features #3: Recent Releases
+[feat-3-recent-releases]: #feat-1-recent-releases
```
Comment (ulrikstrid) [inline rfcs/0003-roadmap-2019.md]: typo: `#feat-1-recent-releases` -> `#feat-3-recent-releases`

File: rfcs/0003-roadmap-2019.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+- Start Date: 2019-07-13
+- Issue/PR: (leave this empty)
+- Point of Contact: Leandro Ostera <leandro@src.technology>
+                    Pontus Nagy <pontusnagy@gmail.com>
+                    Ulrik Strid <ulrik.strid@outlook.com>
+
+# Summary
+[summary]: #summary
+
+This first roadmap is intended to ellaborate on _what_ needs to be done in
```
Comment (cem2ran) [inline rfcs/0003-roadmap-2019.md]:
```suggestion
This first roadmap is intended to elaborate on _what_ needs to be done in
```

File: rfcs/0003-roadmap-2019.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+0. Package Index
+1. Package Search
+2. Package Documentation
+3. Recent Releases
+
+## Features #0: Package Index
+[feat-0-package-index]: #feat-0-package-index
+
+**Definition of Done**
+* We have a index of Reason packages published to a public url
```
Comment (cem2ran) [inline rfcs/0003-roadmap-2019.md]:
```suggestion
* We have an index of Reason packages published to a public url
```

**Review by cem2ran — COMMENTED** [review]
👍 Added a few links and made a few suggestions to fix typos.

**Review by cem2ran — APPROVED** [review]

File: rfcs/0001-reuniverse.md [hunk 8]
```diff
@@ -166,6 +166,9 @@ concerns in a smooth fashion.
    sources of information (such as manually submitted packages), we will build
    an index of reason packages.
 
+   In the future, additional package sources could be included, such as [Opam](https://github.com/ocaml/opam-repository),
```
Comment (p1xelHer0) [inline rfcs/0001-reuniverse.md]: Opam -> opam

File: rfcs/0003-roadmap-2019.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+[feature-1-package-search]: #feature-1-package-search
+
+**Definition of Done**
+* We can search for packages in the index at [Reuniverse](https://reuniverse.github.io/)
+  by name, description, target platform, tags.
+* Searching should handle any amount of packages that we can get in
+  [Feature #0](#feat-0-package-index).
+* Searching should look consistent and beautiful.
+* Clicking on a package will take me to the package documentation.
+* All packages will show where they came from (npm, github, etc)
```
Comment (p1xelHer0) [inline rfcs/0003-roadmap-2019.md]: github -> GitHub or Github?

File: rfcs/0003-roadmap-2019.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+1. Package Search
+2. Package Documentation
+3. Recent Releases
+
+## Feature #0: Package Index
+[feature-0-package-index]: #feature-0-package-index
+
+**Definition of Done**
+* We have an index of Reason packages published to a public url
+* The index is updated periodically.
```
Comment (p1xelHer0) [inline rfcs/0003-roadmap-2019.md]: No period here? We don't use periods in other lists.

File: rfcs/0003-roadmap-2019.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+# Public Phase
+[public-phase]: #public-phase
+
+After we release Reuniverse to the public, we would like some of the
+following to be developed over the rest of the year. The order and priority
+of them is undefined but we will use the input of the community to define it.
+
+## Scan Github for Packages
+Reuniverse should scan Github for packages marked as using the OCaml language
```
Comment (p1xelHer0) [inline rfcs/0003-roadmap-2019.md]: Github -> GitHub? We have to decide on one.

File: rfcs/0003-roadmap-2019.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+that are related to it, across packages, and provide access to the
+documentation of those symbols.
+
+## Documentation Search
+Reuniverse should allow you to search across all the docs for all the packages.
+
+## Playground
+Reuniverse should provide a playground where you can mesh up libraries and
+try them out, provided they are runnable as web projects. This ties up
+closely with work being done by @thangngoc89 on Sketch.sh, and is a point
```
Comment (p1xelHer0) [inline rfcs/0003-roadmap-2019.md]: change this to be a link to https://sketch.sh/? [Sketch.sh](https://sketch.sh/)

**Review by p1xelHer0 — COMMENTED** [review]
These are just some comments about consistent naming:

We should use the same and the "official" name when referring to tools and other projects.
The official site for opam https://opam.ocaml.org/ seems to refer to opam as "opam", all lowercase.
npm https://www.npmjs.com/ refers to npm as "npm" with all lowercase as well.
GitHub seems to refer to themselves as "GitHub" most of the time.

**Review by ulrikstrid — APPROVED** [review]

**Review by p1xelHer0 — APPROVED** [review]

**Pull request merged**